sympy/solvers/ode.py

```diff
@@ -2432,6 +2432,7 @@ def checkodesol(ode, sol, func=None, order='auto', solve_for_func=True):
 
             if sol.lhs == func:
                 s = sub_func_doit(ode_diff, func, sol.rhs)
+                s = simplify(s)
```

Comment:
It is deliberate that we don't use full simplify here. Maybe just use besselsimp if there are bessel functions.

Reply:
okay